tests/includes/juju.sh

```diff
@@ -289,6 +289,8 @@ post_add_model() {
 		add_images_for_vsphere
 		;;
 	esac
+
+	juju set-model-constraints "arch=${BUILD_ARCH:-}"
```

Comment:
It's not really "build arch" though is it? This env var is set in the Makefile when building juju but this is totally unrelated to running tests.
And what if it is not set? Constraints parsing will fail.

Reply:
Taking into consideration your comment below, I will:

1. Check MODEL_ARCH and use it as a model constraint if defined
2. Check WORKLOAD_ARCH and use it as a deployment constraint if defined